### ISSUE TITLE
Checking spelling of abbreviations with single letters and full stops, and word combinations

### ISSUE THREAD

**Issue by Jolar** [SubtitleEdit/subtitleedit#9077]
When checking things like "C.S.I.", the spell checker marks and steps through each letter as a potential mistake, even though "C,S.I." is stored as a name (in either names.xml or en_names_user.xml). Same thing with "Los Angeles" - "Los" and "Angeles" are treated separately and each word becomes marked as an error and needs an input, even though "Los Angeles" is stored as a name.

Is there a possibility to change the spell mechanism to take into account what has been set as a name? To work the same way as it does when changing case?

**Comment by niksedk** [comment]
It works fine here... you might have a broken SE installation? Try with a clean install.
You need to add your abbreviations to the names list.

I've added few abbreviation to SE here: https://github.com/SubtitleEdit/subtitleedit/commit/bc86ab1571343abe4582ad18c98eeb6029f044b9

**Comment by Jolar** [comment]
Hmm. That sound promising! But I have done two re-installs actually, the
first with resetting lists, and the second one clearing all settings. I'll
try deleting things manually and see what happens. Thanks!

Den tis 10 dec. 2024 11:36Nikolaj Olsson ***@***.***> skrev:

**Comment by Jolar** [comment]
Ok, I uninstalled and checked that the folder under Roaming, and the Program folder was gone. Did a new installation but no change. The spell checker really works for you? I think this has been bugging me for years, even on previous computers. See screenshot, this is the first abbreviation I added after the fresh reinstall, "L.D."

https://imgur.com/a/MeNxZtL

**Comment by niksedk** [comment]
Add the "L.D." to the name list via Options - Word lists
Run spell check... all ok.

![image](https://github.com/user-attachments/assets/4f9511d7-aef8-466c-bfd3-f16152cc0948)

**Comment by Jolar** [comment]
I figured it out - if L.D. is the last word on a line without a period it doesn't work. If it's inside a sentence or if there's a period after, like "L.D.." it's all OK. Probably the spell checker always removes the last period because it's usually not related to anything but the structure of the sentence.
Strangely enough this should be completely unrelated to my first issues. After doing a wider check now, everything works fine after the re-install. But as this was the first thing I encountered, I thought there still was issues. Weird...
